contributed_definitions/nyaml/NXpeak_model.yaml

```diff
@@ -0,0 +1,140 @@
+category: base
+doc: |
+  Peak fitting of a one-dimensional spectrum.
+symbols:
+  doc: |
+    The symbols used in the schema to specify e.g. dimensions of arrays.
+  n_peaks: |
+    Number of peaks.
+type: group
+NXpeak_model(NXprocess):
+  region_REGION(NXfit_region):
+    doc: |
+      Subset of the abscisssa that is used for fitting.
+      It is envisioned that regions are labeled as regions_0, regions_1, and so on.
+  peak_PEAK(NXpeak):
+    doc: |
+      One spectral peak of the peak model.
+      It is envisioned that peaks are labeled as peak_0, peak_1, and so on.
+  background(NX_background):
+    doc: |
+      Description of the function used for background subtraction.
+  envelope(NX_NUMBER):
+    unit: NX_ANY
+    doc: |
+      Resulting envelope of all NXpeak instances.
+      
+      Should have the same dimension as the `position` axes of the individual peaks.
+  figure_of_merit(NX_NUMBER):
+    unit: NX_UNITLESS
+    doc: |
+      Figure-of-merit, e.g., :math:`\Chi^2`, or the standard deviation of the residuals,
+      to determine the goodness of fit, i.e., how well the peak model fits the measured
+      observations.
+      
+      This value (which is a single number) is oftenused to guide adjustments to the
+      fitting parameters in the peak fitting process.
+  fitting_method:
+    doc: |
+      Description of the method used to optimize the parameters during peak fitting.
+      Examples:
+      - least squares
+      - non-linear least squares
+      - Levenberg-Marquardt algorithm (damped least-squares)
+      - linear regression
+      - Bayesian linear regression
+  relative_concentration(NX_FLOAT):
+    unit: NX_ANY
+    doc: |
+      Atomic concentration of each species defined by one peak in the peak model.
+
```

Comment:
Missing: Place to put the dependent and independent data, the resulting fit function, and the residuals. Maybe define an NXdata here to hold these, then they can be automatically plotted (maybe also with the individual peak contributions).

Reply:
Now the dependent/independent, the envelope, and the residuals are part of an NXdata group.